tests/storage/datavolume.go

```diff
@@ -1179,33 +1171,74 @@ var _ = Describe(SIG("DataVolume Integration", func() {
 			}
 			_, err := virtClient.VirtualMachinePreference(testsuite.GetTestNamespace(nil)).Create(context.Background(), virtualMachinePreference, metav1.CreateOptions{})
 			Expect(err).ToNot(HaveOccurred())
-
-			// Bind VirtualMachinePreference to VirtualMachine
-			vm.Spec.Preference = &v1.PreferenceMatcher{
-				Name: virtualMachinePreference.Name,
-				Kind: instancetypeapi.SingularPreferenceResourceName,
-			}
 		})
 
 		AfterEach(func() {
 			Expect(virtClient.VirtualMachinePreference(virtualMachinePreference.Namespace).Delete(context.Background(), virtualMachinePreference.Name, metav1.DeleteOptions{})).To(Succeed())
 		})
 
-		It("should use PreferredStorageClassName when storage class not provided by VM", func() {
-			// Remove storage class name from VM definition
-			vm.Spec.DataVolumeTemplates[0].Spec.Storage.StorageClassName = nil
+		Context("using storage API", func() {
```

Comment:
supernit - This could all just be two tables for the two test cases (assert preference value is used and not used) but I appreciate that's awkward to write thus the supernit, ignore if you want.